src/charmed_kubeflow_chisme/testing/cos_integration.py

```diff
@@ -201,6 +192,24 @@ def _get_alert_rules(data: str) -> Set[str]:
     return {alert_rules["alert"]}
 
 
+def _get_templates(data: str) -> Set[Dict[str, dict]]:
+    """Get all templates from relation data, where it's defined as string.
+
+    Example of relations data of grafana-dashboard would be:
+
+    ```Python
+    'dashboards': '{"templates": {"file:jupyter-notebook-controller.json": {"content": ...
+    ```
+    """
+    templates_raw = yaml.safe_load(data).get("templates", {})
+    templates = {}
+    for key, value in templates_raw.items():
+        file_name = key.replace("file:", "")  # template key is defined as 'file:<file_name>'
+        templates[file_name] = {"charm": value["charm"], "juju_topology": value["juju_topology"]}
+
+    return templates
```

Comment:
Should the function name reflect that this doesn't just fetch the templates but also modifies before returning them?

Reply:
I think it the doctstring at least should reflect how this works